File: examples/python/playground/triks/@demo/hello-py/graph.py
"""
Hello Python Trik - A simple Python trik demonstrating native execution.

This trik runs directly in-process within the Python gateway.
"""

from __future__ import annotations

import sys
from typing import Any


class HelloPythonGraph:
    """A simple Python trik that demonstrates native Python execution."""

    async def invoke(self, input_data: dict[str, Any]) -> dict[str, Any]:
        """
        Main invoke method - called by the TrikHub gateway.

        Args:
            input_data: The input from the gateway containing action and input.

        Returns:
            The result with responseMode and agentData.
        """
        action = input_data.get("action")
        action_input = input_data.get("input", {})

        if action == "greet":
            return self.greet(action_input)
        elif action == "reverse":
            return self.reverse(action_input)
        else:
            return {
                "responseMode": "template",
                "agentData": {
                    "template": "error",
                    "message": f"Unknown action: {action}",
                },
            }

    def greet(self, input_data: dict[str, Any]) -> dict[str, Any]:
        """Generate a greeting message."""
        name = input_data.get("name")

        if not name or not isinstance(name, str):
            return {
                "responseMode": "template",
                "agentData": {
                    "template": "error",
                    "message": "Name is required and must be a string",
                },
            }

        greeting = f"Hello, {name}! Welcome to TrikHub from Python!"
        python_version = f"Python {sys.version_info.major}.{sys.version_info.minor}"

        return {
            "responseMode": "template",
            "agentData": {
                "template": "success",
                "message": greeting,
                "language": python_version,
            },
        }

    def reverse(self, input_data: dict[str, Any]) -> dict[str, Any]:
        """Reverse a string."""
        text = input_data.get("text")

        if not text or not isinstance(text, str):
            return {
                "responseMode": "template",
                "agentData": {
                    "template": "error",
                    "message": "Text is required and must be a string",
                },
            }

        reversed_text = text[::-1]

        return {
            "responseMode": "template",
            "agentData": {
                "template": "result",
                "original": text,
                "reversed": reversed_text,
            },
        }


# Export the graph instance
graph = HelloPythonGraph()

File: examples/python/playground/triks/@demo/hello-js/graph.js
/**
 * Hello JS Trik - A simple JavaScript trik demonstrating cross-language execution.
 *
 * This trik runs via the Node.js worker subprocess when executed from the Python gateway.
 */

class HelloGraph {
  /**
   * Main invoke method - called by the TrikHub gateway.
   *
   * @param {Object} input - The input from the gateway
   * @param {string} input.action - The action to execute
   * @param {Object} input.input - The action input
   * @returns {Object} The result
   */
  async invoke(input) {
    const { action, input: actionInput } = input;

    switch (action) {
      case "greet":
        return this.greet(actionInput);
      case "calculate":
        return this.calculate(actionInput);
      default:
        return {
          responseMode: "template",
          agentData: {
            template: "error",
            message: `Unknown action: ${action}`,
          },
        };
    }
  }

  /**
   * Generate a greeting message.
   */
  greet(input) {
    const { name } = input;

    if (!name || typeof name !== "string") {
      return {
        responseMode: "template",
        agentData: {
          template: "error",
          message: "Name is required and must be a string",
        },
      };
    }

    const greeting = `Hello, ${name}! Welcome to TrikHub from JavaScript!`;
    const timestamp = new Date().toISOString();

    return {
      responseMode: "template",
      agentData: {
        template: "success",
        message: greeting,
        timestamp: timestamp,
      },
    };
  }

  /**
   * Perform a simple calculation.
   */
  calculate(input) {
    const { operation, a, b } = input;

    if (typeof a !== "number" || typeof b !== "number") {
      return {
        responseMode: "template",
        agentData: {
          template: "error",
          message: "Both 'a' and 'b' must be numbers",
        },
      };
    }

    let result;
    let opName;

    switch (operation) {
      case "add":
        result = a + b;
        opName = `${a} + ${b}`;
        break;
      case "subtract":
        result = a - b;
        opName = `${a} - ${b}`;
        break;
      case "multiply":
        result = a * b;
        opName = `${a} × ${b}`;
        break;
      case "divide":
        if (b === 0) {
          return {
            responseMode: "template",
            agentData: {
              template: "error",
              message: "Cannot divide by zero",
            },
          };
        }
        result = a / b;
        opName = `${a} ÷ ${b}`;
        break;
      default:
        return {
          responseMode: "template",
          agentData: {
            template: "error",
            message: `Unknown operation: ${operation}`,
          },
        };
    }

    return {
      responseMode: "template",
      agentData: {
        template: "result",
        result: result,
        operation: opName,
      },
    };
  }
}

// Export the graph instance
const graph = new HelloGraph();
module.exports = { graph };
